comp/core/tagger/tagger.go

```diff
@@ -365,8 +367,14 @@ func (t *TaggerClient) ResetCaptureTagger() {
 func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
 	cardinality := taggerCardinality(originInfo.Cardinality)
 
+	// If dogstatsdUnifiedEnabled is disabled, we use DogStatsD's Legacy Origin Detection.
+	// TODO: remove this when dogstatsd_origin_detection_unified is enabled by default
+	if !t.datadogConfig.originDetectionUnifiedEnabled {
```

Comment:
I'm wondering if the test should be `if !t.datadogConfig.originDetectionUnifiedEnabled && originInfo.ProductOrigin == taggertypes.ProductOriginDogStatsD`

Reply:
Yes that would be better 👍 
https://github.com/DataDog/datadog-agent/pull/24670/commits/3b04f2a2e7e606e7ca2a737e231a2e3b8dfd7b59